js-core/homeworks/homework-6/src/main.js

```diff
@@ -0,0 +1,503 @@
+    },
+    {
+      name: 'Hammond Harris',
+      id: 46,
+    },
+    {
+      name: 'Ester Hamilton',
+      id: 47,
+    },
+    {
+      name: 'Lopez Hopper',
+      id: 48,
+    },
+    {
+      name: 'Newton Cantrell',
+      id: 49,
+    },
+  ],
+  range: [0, 1, 2, 3, 4, 5, 6, 7, 8, 9],
+  tags: ['nulla', 'Lorem', 'labore', 'dolore', 'enim'],
+  longitude: '0.298835',
+  latitude: '20.822214',
+  registered: 'Saturday, July 2, 2016 10:52 PM',
+  about: 'Ipsum adipisicing nulla adipisicing culpa qui et esse labore ut nostrud excepteur amet elit voluptate. Ut ex laborum enim sit. Proident eu elit tempor occaecat dolor.',
+  address: '597 Agate Court, Interlochen, Utah, 6828',
+  phone: '+1 (987) 453-2740',
+  email: 'lenora.barr@uberlux.me',
+  company: 'UBERLUX',
+  name: {
+    last: 'Barr',
+    first: 'Lenora',
+  },
+  eyeColor: 'brown',
+  age: 37,
+  picture: 'http://placehold.it/32x32',
+  balance: '$3,110.45',
+  isActive: true,
+  guid: '74d9074f-004a-447f-8721-12d884011852',
+  index: 0,
+  _id: '584bac50967253d4322db586',
+};
+
+/*
+ *
+ * TASK 1
+ * Посчитайте количество букв а, в передаваемом параметре
+ *
+ * */
+
+let randomString = 'aaa b a w c ';
+
+let user = {
+  name: 'Albina',
+};
+
+let javaScript = {
+  html: 'JavaScript',
+};
+
+function countLetterA(str) {
+    let counterA = 0;
+    let strTransformToArray = str.split('');
+    strTransformToArray.forEach(function(value, index, arr) {
+        if(value === 'a') {
+            counterA += 1;
+        }
+    });
+    return counterA;
+}
+
+console.log(countLetterA(randomString)); // 4
+console.log(countLetterA(user.name + javaScript.html)); // 3
+
+// ! ! ! ! ! ! ! ! ! ! !
+/*
+ *
+ * TASK 2
+ * Вам дано предложение, вам необходимо перевернуть каждое
+ * слово в предложении в обратном порядке
+ *
+ * */
+
+function reverseEachWord() {
+}
+
+function reverseEachWord(str) {
+    let newArray = [];
+    let strTransformToArray = str.split(' ');
+    strTransformToArray.forEach(function(value, index, arr) {
+        let reversingEachWord = value.split('').reverse().join('');
+        newArray.push(reversingEachWord);
+    });
+    let newArrayTransformToString = newArray.join(' ');
+    return newArrayTransformToString;
+}
+
+
+console.log(reverseEachWord('You don\'t have to do anything special to begin using the DOM. Different browsers have different implementations of the DOM'));
+// uoY t'nod evah ot od gnihtyna laiceps ot nigeb gnisu eht .MOD tnereffiD sresworb evah tnereffid snoitatnemelpmi fo eht MOD
+
+console.log(reverseEachWord('The Document Object Model (DOM) is a programming interface for HTML and XML documents'));
+// ehT tnemucoD tcejbO ledoM )MOD( si a gnimmargorp ecafretni rof LMTH dna LMX stnemucod
+
+/*
+ * TASK 3
+ * Добавьте в функцию reverseEachWord второй параметр,
+ * данный параметр булево, если true - тогда предложение так же
+ * переворачиваются в обратном порядке
+ * */
+
+function reverseEachWord2(str, boolean) {
+    if(boolean === true) {
```

Comment:
hm, interesting solution.

But I would name boolean better, with context what does it mean for example

**shouldSentceBeReversed**